backend/src/main/java/rfid/RfidTagUid.java

```diff
@@ -25,7 +23,7 @@ public RfidTagUid(byte[] uid) {
      */
     public RfidTagUid(String uid) {
         if (uid.length() != RFID_TAG_UID_STRING_LENGTH) {
-            throw new GeneralException(String.format("Got RFID UID with unexpected length: '%s'", uid.length()));
+            throw new RuntimeException(String.format("Got RFID UID with unexpected length: '%s'", uid.length()));
```

Comment:
I would say this is a good candidate for an `IllegalArgumentException`, since the provided `uid` has an unexpected format.

Reply:
Good comment! Fixed!